<?xml version="1.0"?>

<!-- ackermann_vehicle.launch

Use Gazebo to simulate a vehicle with Ackermann steering.

Arguments:
    namespace (string, default: ackermann_vehicle)
        Vehicle namespace.
    world_name (string, default: worlds/empty_sensor.world)
        Gazebo world name.
    cmd_timeout (float, default: 0.5)
        Command timeout passed to the ackermann_controller.py node.
    x (float, default: 0.0)
    y (float, default: 0.0)
    z (float, default: 0.1)
    roll (float, default: 0.0)
    pitch (float, default: 0.0)
    yaw (float, default: 0.0)
        Vehicle pose. x, y, and z are measured in meters. roll, pitch, and yaw
        are measured in radians.

Copyright (c) 2013 Wunderkammer Laboratory

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->

<launch>

  <!--arg name="namespace"   default="QCar"/-->
  <!--arg name="world"       default="empty" doc="worlds: [empty, warehouse, checkerboard]"/-->
  <!--arg name="cmd_timeout" default="0.5"/-->
  <!--arg name="rectify"     default="true" /-->
  <!--arg name="stereo_view" default="false" /-->
  <!--empty_sensor; obstacle_sensor; empty--> 
  <!--arg name="paused" default="false"/-->
  <!--arg name="use_sim_time" default="true"/-->
  <!--arg name="gui" default="false"-->
  <!--arg name="headless" default="false"/-->
  <!--arg name="debug" default="false"/-->
  <arg name="rviz"       default="false"/>
  <arg name="ns1"        default="qcar1"/>
  <arg name="ns2"        default="qcar2"/>
  <arg name="ns3"        default="qcar3"/>
  <arg name="ns4"        default="qcar4"/>


  <arg name="cmd_timeout" default="0.5"/>
  <arg name="world" default="empty_sensor"/>


  <!-- Vehicle pose -->
  <arg name="x"           default="0.0"/>
  <arg name="y"           default="0.0"/>
  <arg name="z"           default="0.2"/>
  <arg name="roll"        default="0.0"/>
  <arg name="pitch"       default="0.0"/>
  <arg name="yaw"         default="0.0"/>

  <!-- <group ns="$(arg namespace)"> -->
    <!-- Create the world. -->
    <!-- <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="debug" value="$(arg debug)" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="$(arg use_sim_time)"/>
        <arg name="headless" value="$(arg headless)"/>
    </include> -->
    <!-- Spawn the vehicle. -->
    <!-- Spawn the vehicle. -->

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <!--arg name="use_sim_time" value="true"/-->
    <arg name="gui"          value="false"/>	
    <arg name="world_name" value="$(find qcar)/worlds/$(arg world).world"/>
  </include> 

  <node name="distance_node" pkg="qcar" type="distancenode.py"/> 

  <!-- Spawn the robots. -->

  <group ns="$(arg ns1)">
  <!--param name="tf_prefix" value="$(arg ns1)"/-->
    <include file="$(find qcar)/launch/qcar_base_MPC.launch">
      <arg name="ns"            value="$(arg ns1)"/>
      <arg name="x"             value="0.18"/>
      <arg name="y"             value="-1.16"/>
      <arg name="yaw"           value="1.5707963268"/>
    </include>
  </group>

  <group ns="$(arg ns2)">
    <!--param name="tf_prefix" value="$(arg ns2)"/-->
    <include file="$(find qcar)/launch/qcar_base_MPC.launch">
      <arg name="ns"            value="$(arg ns2)"/>
      <arg name="x"             value="-1.16"/>
      <arg name="y"             value="-0.18"/>
      <arg name="yaw"           value="0.0"/>
    </include>
  </group>

  <group ns="$(arg ns3)">
    <!--param name="tf_prefix" value="$(arg ns3)"/-->
    <include file="$(find qcar)/launch/qcar_base_MPC.launch">
      <arg name="ns"            value="$(arg ns3)"/>
      <arg name="x"             value="1.16"/>
      <arg name="y"             value="0.18"/>
      <arg name="yaw"           value="3.1415926536"/>
    </include>
  </group>

  <group ns="$(arg ns4)">
    <!--param name="tf_prefix" value="$(arg ns4)"/-->
    <include file="$(find qcar)/launch/qcar_base_MPC.launch">
      <arg name="ns"            value="$(arg ns4)"/>
      <arg name="x"             value="-0.18"/>
      <arg name="y"             value="1.16"/>
      <arg name="yaw"           value="-1.5707963268"/>
    </include>
  </group>


  <include file="$(find waypoint_loader)/launch/waypoint_loader.launch"/>
  <include file="$(find waypoint_updater)/launch/waypoint_updater.launch"/>
</launch>
